tests/manage/mcg/test_bucket_replication.py

```diff
@@ -368,6 +409,76 @@ def test_unidirectional_bucketclass_replication(
+            command=f"psql -h 127.0.0.1 -p 5432 -U postgres -d nbcore -c "
+            f"\"SELECT data ->> 'replication_policy_id' FROM buckets WHERE data ->> 'name'='{source_bucket_name}';\"",
+            out_yaml_format=False,
+        ).split("\n")[2]
+
+        # write random objects to the bucket
+        write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            source_bucket_name,
+            test_directory_setup.origin_dir,
+            amount=5,
+            mcg_obj=mcg_obj_session,
+        )
+
+        if target_bucketclass is None:
+            # Delete the target bucket
+            target_bucket.delete()
+            logger.info(f"Deleted target bucket {target_bucket_name}")
+
+            def check_nb_endpoint_logs():
+                for pod in get_noobaa_endpoint_pods():
+                    if (
+                        "The AWS access key Id you provided does not exist in our records"
+                        in get_pod_logs(pod.name)
+                    ):
+                        return True
+                return False
+
+            sample = TimeoutSampler(timeout=360, sleep=10, func=check_nb_endpoint_logs)
+            assert sample.wait_for_func_status(
+                result=False
+            ), "Error 'The AWS access key Id you provided does not exist in our records' found in the endpoint logs"
+        else:
+            # delete the bucket
+            source_bucket.delete()
+            logger.info(f"Deleted source bucket {source_bucket_name}")
+
+            # check in db that the replication config was deleted
+            replication_conf_count = nb_db_pod.exec_cmd_on_pod(
+                command=f"psql -h 127.0.0.1 -p 5432 -U postgres -d nbcore -c "
+                f"\"SELECT COUNT (*) FROM replicationconfigs WHERE _id='{replication_id}'\"",
+                out_yaml_format=False,
+            ).split("\n")[2]
+
+            assert (
+                int(replication_conf_count) == 0
+            ), f"Replication config for {source_bucket_name} is not deleted!"
+
+            # check in noobaa core logs
+            def check_noobaa_core_logs():
+                if "find_rules_updated_longest_time_ago:  []" in get_pod_logs(
+                    get_noobaa_core_pod().name
+                ):
+                    return True
+                else:
+                    return False
+
+            logger.info(
+                f"Checking the noobaa core logs to see if the replication has "
+                f"stopped from {source_bucket_name} to {target_bucket_name}"
+            )
+            sample = TimeoutSampler(timeout=900, sleep=10, func=check_noobaa_core_logs)
+            assert sample.wait_for_func_status(
+                result=True
+            ), f"Replication hasn't stopped between {source_bucket_name} to {target_bucket_name}"
+            logger.info(
+                f"Replication has stopped from {source_bucket_name} to {target_bucket_name}"
+            )
+
```

Comment:
We should move this under a separate test method - even if it means to copy the setup from `test_unidirectional_bucketclass_replication`. 

The original method has about 30 lines of code, while the coverage of this BZ adds about 80 that are only specific to one of the params (`AWStoDefault-BS-Hybrid`). I think this justifies the separation for the sake of readability and maintainability.

Reply:
@sagihirshfeld The reason why i added this coverage to the existing code is to have this coverage for all the type of params not just `AWStoDefault-BS-Hybrid`. we check replication has stopped for all the types of buckets here.

Reply:
Oh I must've missed the "else" part here in the original review, this is ok then.